app/src/main/java/com/sunasterisk/travelapp/ui/main/MainActivity.kt

```diff
@@ -1,12 +1,55 @@
 package com.sunasterisk.travelapp.ui.main
 
-import androidx.appcompat.app.AppCompatActivity
-import android.os.Bundle
+import androidx.appcompat.app.ActionBarDrawerToggle
+import androidx.core.view.GravityCompat
+import androidx.fragment.app.Fragment
+import androidx.viewpager.widget.ViewPager
 import com.sunasterisk.travelapp.R
+import com.sunasterisk.travelapp.ui.base.BaseActivity
+import com.sunasterisk.travelapp.ui.home.HomeFragment
+import com.sunasterisk.travelapp.utils.replaceFragmentInActivity
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.include_home_view.*
 
-class MainActivity : AppCompatActivity() {
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+class MainActivity : BaseActivity(), HomeFragment.OnTabsSetupListener {
+
+    override val layoutResource get() = R.layout.activity_main
+
+    override fun initComponent() {
+        replaceFragmentInActivity(HomeFragment.newInstance(), R.id.frameMain)
+        initToolbar()
+        initNavigationMenu()
+    }
+
+    override fun onAttachFragment(fragment: Fragment) {
+        if (fragment is HomeFragment) {
+            fragment.setTabsCallbackListener(this)
+        }
+    }
+
+    private fun initToolbar() {
+        setSupportActionBar(toolbar)
+        actionBar?.apply {
+            setDisplayHomeAsUpEnabled(true)
+            setHomeButtonEnabled(true)
+        }
+    }
+
+    private fun initNavigationMenu() {
+        ActionBarDrawerToggle(
+            this,
+            drawerMain,
+            toolbar,
+            R.string.navigation_drawer_open,
+            R.string.navigation_drawer_close
+        ).apply {
+            syncState()
+            drawerMain.setDrawerListener(this)
+        }
```

Comment:
thêm biến phụ để clear hơn đi ông 
```suggestion
        val actionBarDrawerTogger = ActionBarDrawerToggle(
            this,
            drawerMain,
            toolbar,
            R.string.navigation_drawer_open,
            R.string.navigation_drawer_close
        )
```

Reply:
ok ông